Libraries/LibGUI/TextBox.cpp

```diff
@@ -37,4 +37,67 @@ TextBox::~TextBox()
 {
 }
 
+void TextBox::keydown_event(GUI::KeyEvent& event)
+{
+    TextEditor::keydown_event(event);
+
+    if (event.key() == Key_Up) {
+        if (on_up_pressed)
+            on_up_pressed();
+
+        if (!m_history_enabled || !has_any_history() || !can_go_backwards_in_history())
+            return;
+
+        if (m_history_index >= static_cast<int>(m_input_history.size()))
+            m_saved_input = text();
+
+        m_history_index--;
+        set_text(current_historical_input());
+    } else if (event.key() == Key_Down) {
+        if (on_down_pressed)
+            on_down_pressed();
+
+        if (!m_history_enabled || !has_any_history())
+            return;
+
+        if (can_go_forwards_in_history()) {
+            m_history_index++;
+            set_text(current_historical_input());
```

Comment:
This feels muddy. It's not obvious that incrementing `m_history_index` causes `current_historical_input` to return a different value.
How about:
```cpp
++m_history_index;
set_text(m_input_history[m_history_index]);
```